include/emp/prefab/ToggleButtonGroup.hpp

```diff
@@ -0,0 +1,243 @@
+      web::internal::DivInfo * info_ref
+    ) : ButtonGroup(info_ref)
+    {
+      AddAttr(
+        "class", "btn-group-toggle", "data-toggle", "buttons"
+      );
+
+      web::Element activate_label("label", emp::to_string(GetID(), "_activate"));
+      web::Element deactivate_label("label", emp::to_string(GetID(), "_deactivate"));
+      *this << activate_label;
+      *this << deactivate_label;
+
+      if (!cassette_style) {
+        AddAttr("class", "hide_inactive");
+      } else if (grayout) {
+        AddAttr("class", "grayout");
+      }
+
+      auto & on_toggle = GetCallback();
+
+      activate_label.AddAttr(
+        "class", "btn",
+        "class", emp::to_string("btn-outline-", activate_style)
+      );
+      activate_label.OnClick([tog=*this, &handle_toggle=on_toggle]() mutable {
+        tog.SetActive(true);
+        handle_toggle(true);
+      });
+      // OnClick used due to a strange bug(?) in which the input radios only
+      // fire their onchange function once
+      // Probably due to Bootstrap/jQuery weirdness and this is a last minute
+      // work around ¯\_(ツ)_/¯
+
+      web::Input activate_radio(
+        [](std::string){ ; },
+        "radio", "", emp::to_string(GetID(), "_activate_radio"),
+        false, false
+      );
+      activate_label << activate_radio;
+      activate_label << std::forward<L1_TYPE>(activate_indicator);
+
+      deactivate_label.AddAttr(
+        "class", "active",
+        "class", "btn",
+        "class", emp::to_string("btn-outline-", deactivate_style)
+      );
+      deactivate_label.OnClick([tog=*this, &handle_toggle=on_toggle]() mutable {
+        tog.SetActive(false);
+        handle_toggle(false);
+      });
+
+      web::Input deactivate_radio(
+        [](std::string){ ; },
+        "radio", "", emp::to_string(GetID(), "_deactivate_radio"),
+        false, true
+      );
+      deactivate_label << deactivate_radio;
+      deactivate_label << std::forward<L2_TYPE>(deactivate_indicator);
+    }
+
+    public:
+    /**
+     * @param activate_indicator a string, FontAwesomeIcon or other component
+     * indicating that the first button actives this toggle
+     * @param deactivate_indicator a string, FontAwesomeIcon or other component
+     * indicating that the second button deactivates this toggle
+     * @param activate_style a bootstrap style (primary, secondary, etc) for
+     * the first button
+     * @param deactivate_style a bootstrap style (primary, secondary, etc) for
+     * the second button
+     * @param cassette_style whether the toggle should display in cassette style
+     * (both buttons visible) or do a swap on toggle (one button visible)
+     * @param grayout in cassette mode, whether buttons should be
+     * grayed out to further emphasize the current state
+     * @param in_id HTML ID of ToggleButtonGroup div
+     */
+    template<typename L1_TYPE, typename L2_TYPE>
+    ToggleButtonGroup(
+      L1_TYPE && activate_indicator,
```

Comment:
Hmm... can we provide reasonable defaults as default params? Might be tricky with the templated constructor...

Failing that I would add two static makers, one called MakeCassetteToggleGroup and one called MakeSwapToggleButtonGroup that return a ToggleButtonGroup with all these things filled in reasonably

Reply:
Not sure what you mean here. I already gave most parameters default values, but I could move some of these methods from the constructor to separate setter methods if the constructor is overwhelming.